### ISSUE TITLE
fix: "Show Widget Inspector" resets the "scoped" state.

### ISSUE THREAD

**Issue by yehorh** [jinyus/lite_ref#22]
**Description**

When developers enable or disable the Widget Inspector, scoped LiteRefs are recreated.

**Steps To Reproduce**

```dart
import 'package:flutter/material.dart';
import 'package:state_beacon/state_beacon.dart';

void main() {
  runApp(LiteRefScope(child: const MyApp()));
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: const LiteRefPage(),
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
    );
  }
}

class LiteRefPage extends StatelessWidget {
  const LiteRefPage({super.key});

  @override
  Widget build(BuildContext context) {
    final myInt = stateRef.of(context);

    return Scaffold(
      floatingActionButton: FloatingActionButton(
        onPressed: () {
          myInt.value++;
        },
        child: const Icon(Icons.add),
      ),
      appBar: AppBar(
        title: const Text('LiteRef'),
        actions: [
          IconButton(
            icon: const Icon(Icons.search_rounded),
            onPressed: () {
              WidgetsApp.debugShowWidgetInspectorOverride =
                  !WidgetsApp.debugShowWidgetInspectorOverride;
            },
          ),
        ],
      ),
      body: Center(
        child: Text(
          'int: ${myInt.watch(context)}',
          style: Theme.of(context).textTheme.titleLarge,
        ),
      ),
    );
  }
}

final stateRef = Ref.scoped(
  (context) => Beacon.writable(0),
  dispose: (beacon) => beacon.dispose(),
);
```

https://github.com/jinyus/lite_ref/assets/10909216/c24dcf84-cf52-4464-bd6b-933cc044ea8d

**Comment by zupat** [comment]
Thanks for the reporting this. I figured out the bug and working on a PR

**Comment by zupat** [comment]
Fixed in 0.7.0

LMK if everything is good on your end.

**Comment by yehorh** [comment]
I have overridden dependencies, and now everything is great! 
Thanks a lot. 
It also fixed another bug with multiple recreations of state during the build phase that I hadn't described yet.
